homeassistant/components/airtouch5/cover.py

```diff
@@ -0,0 +1,139 @@
+"""Representation of the Damper for AirTouch 5 Devices."""
+import logging
+from typing import Any
+
+from airtouch5py.airtouch5_simple_client import Airtouch5SimpleClient
+from airtouch5py.packets.zone_control import (
+    ZoneControlZone,
+    ZoneSettingPower,
+    ZoneSettingValue,
+)
+from airtouch5py.packets.zone_name import ZoneName
+from airtouch5py.packets.zone_status import ZoneStatusZone
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from . import Airtouch5ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import Airtouch5Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: Airtouch5ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Airtouch 5 Cover entities."""
+    client = config_entry.runtime_data
+
+    entities: list[CoverEntity] = []
+
+    # Each zone has a cover for its open percentage
+    for zone in client.zones:
+        entities.append(
+            Airtouch5ZoneOpenPercentage(
+                client, zone, client.latest_zone_status[zone.zone_number].has_sensor
+            )
+        )
+
+    async_add_entities(entities)
```

Comment:
This can be combined with a generator expression 

```suggestion
    async_add_entities(Entity() for X in client.zones)
```

Reply:
I'm not overly familiar with python but from what I understand, your suggestion won't work as-is. Can you elaborate on what you want me to change?

Reply:
updated